### PR TITLE
Kafka SaslOptions

### LINKED ISSUE #5: Provide example for Kafka authentication
Hi!

My credentials/identifiers are like this and I wonder how to pass such credentials to the Kafka crate. I could not find such options in the docs.

```
ssl.endpoint.identification.algorithm=https
security.protocol=SASL_SSL
sasl.jaas.config=org.apache.kafka.common.security.plain.PlainLoginModule required username="{{ CLUSTER_API_KEY }}" password="{{ CLUSTER_API_SECRET }}";
```

Could you please outline how to connect with such config attributes?

Thank you!

### PR DESCRIPTION
**Pull request by tyt2y3**
Fixes #5

### CODE REVIEW

File: sea-streamer-kafka/src/streamer.rs [hunk 1]
```diff
@@ -213,3 +320,25 @@ fn create_admin(
...
+                .username("uu")
+                .password("pp"),
+        );
+        let mut client_config = ClientConfig::new();
+        options.make_client_config(&mut client_config);
+        assert_eq!(client_config.get("security.protocol"), Some("SASL_SSL"));
+        assert_eq!(client_config.get("sasl.mechanism"), Some("PLAIN"));
+        assert_eq!(client_config.get("sasl.username"), Some("uu"));
+        assert_eq!(client_config.get("sasl.password"), Some("pp"));
+    }
```
Comment (tyt2y3) [inline sea-streamer-kafka/src/streamer.rs:343]: .

**Comment by tyt2y3** [comment]
Does this work for you? @hgzimmerman

**Comment by hgzimmerman** [comment]
As far as SASL goes, it looks good.

My example had an additional `bootstrap.servers` option set, which I'm unsure if is needed for SASL authentication to work. (The bootstrap server(s) provides metadata used to route a client to a particular node(s) to connect to for a topic, among other things). I'm unsure how commonly used that option is or if it is worth making a special function for it.

I think if that proves to be necessary, its best followed up on separately with a `set_bootstrap_server(&str)` function.

**Comment by tyt2y3** [comment]
Thanks. Actually the "bootstrap.servers" key is set for every connection with the values coming from `StreamerUri`. May be your bootstrap.servers string is not compatible with `Url`?

https://github.com/SeaQL/sea-streamer/blob/5f552b69aee74f5c4bc89d82161a919b6c1cbb97/sea-streamer-kafka/src/consumer.rs#L646

**Comment by hgzimmerman** [comment]
No, my connection profile was just copied nearly verbatim from  https://github.com/confluentinc/librdkafka/issues/4119#issuecomment-1361958501 and I wasn't aware that the config option was already set elsewhere in sea-streamer-kafka. 

This is good news to me, in that I get to remove some redundant code and config values.

**Comment by tyt2y3** [comment]
Cool! I am going to make a patch release this week end.

**Comment by tyt2y3** [comment]
Released in https://github.com/SeaQL/sea-streamer/releases/tag/0.2.1

**Pull request merged**